### PR TITLE
feat: Update existing configs with missing package-ecosystems

### LINKED ISSUE #108: Detect new ecosystems
### Is your feature request related to a problem?

If you have a `dependabot.yml` with config for e.g. ecosystem `maven`, then add a new ecosystem in the same project (i.e. `npm`), then evergreen won't try to update `dependabot.yml` to include `npm`.

### Describe the solution you'd like

A new PR/Issue should be created when a project contains a `dependabot.yml`-file, but with missing ecosystems.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by zkoppert**
# Pull Request

fixes #108 

## Proposed Changes

This pull request enhances the functionality of the Dependabot configuration file builder and environment variables handler in order to allow the action to recognize and add to existing dependabot configurations instead of skipping over those repos.

The changes include
- Addition of a new parameter to the `build_dependabot_file` function to handle existing configurations,
- Creation of a new function to add existing ecosystems to the exempt list
- Introduction of a new environment variable to decide whether to update existing configurations. Set to False by default to not disrupt current users.
- New tests that cover 2 scenarios
  - `dependabot.yml` exists and has all needed package-ecosystems already in it
  - `dependabot.yml` exists and has 1 ecosystem in it and another ecosystem that should be added

### Author/Contributor

- [x] If documentation is needed for this change, has that been included in this pull request
- [x] run `make lint` and fix any issues that you have introduced
- [x] run `make test` and ensure you have test coverage for the lines you are introducing

### Reviewer

- [x] Label as either `fix`, `documentation`, `enhancement`, `infrastructure`, `maintenance` or `breaking`

### CODE REVIEW

File: dependabot_file.py [hunk 1]
```diff
@@ -305,3 +315,15 @@ def build_dependabot_file(repo, group_dependencies, exempt_ecosystems) -> str |
...
+def add_existing_ecosystem_to_exempt_list(exempt_ecosystems, existing_config):
+    """
+    Add the existing package ecosystems found in the dependabot.yml
+    to the exempt list so we don't get duplicate entries and maintain configuration settings
+    """
+    if existing_config:
+        existing_config_obj = yaml.safe_load(existing_config.decoded)
+        if existing_config_obj:
+            for entry in existing_config_obj.get("updates", []):
+                exempt_ecosystems.append(entry["package-ecosystem"])
```
Comment (zkoppert) [inline dependabot_file.py:329]: Is it okay to modify the `exempt_ecosystems` list? Seemed convenient, but not sure if creating a copy would have been better so as to preserve the original input? Overthinking or good idea?

**Comment by jmeridth** [comment]
@zkoppert still looking at this.  Awesome addition 🎉 

Could you please update the `.env-example` with the new environment variable?

**Comment by zkoppert** [comment]
Great catch on the `.env-example`! Added that ✅

File: dependabot_file.py (hunk 1 above)
Comment (jmeridth) [reply dependabot_file.py:329]: If we don't need the original, I'm good overwriting.

File: evergreen.py [hunk 2]
```diff
@@ -77,20 +78,29 @@ def main():  # pragma: no cover
...
             print("Skipping " + repo.full_name + " (visibility-filtered)")
             continue
+        existing_config = None
         try:
-            if repo.file_contents(".github/dependabot.yml").size > 0:
-                print(
-                    "Skipping " + repo.full_name + " (dependabot file already exists)"
-                )
-                continue
+            existing_config = repo.file_contents(".github/dependabot.yml")
```
Comment (jmeridth) [inline evergreen.py:83]: Extract to function with file name as argument?

since exactly like changes below except file name?

**Review by jmeridth — APPROVED** [review]
Great change. Couple suggestions but not blockers.

File: evergreen.py (hunk 2 above)
Comment (zkoppert) [reply evergreen.py]: [386e781](https://github.com/github/evergreen/pull/140/commits/386e78180aaf087e191e745063474d6e25748053)

**Comment by vidwah-nte** [comment]
Thank you for adding this feature.
I briefly looked through the changes, and it's not apparent to me how a project can mark that a ecosystem _shouldn't_ be added to the dependabot-file. You can exclude ecosystems globally through the `exempt_repos` config, but can it be done per project?

**Comment by zkoppert** [comment]
Great point @vidwah-nte, I've opened a new issue to track that request as currently it is not supported. https://github.com/github/evergreen/issues/151

**Pull request merged**